### PR TITLE
refactor(nc-gui): update useCellUrlConfig & remove overlay injection

### PR DESCRIPTION
**Pull request by bcakmakoglu**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by bcakmakoglu** [comment]
@dolsem I haven't checked if the functionality works 100% correctly now (I am swamped in work, so I just changed the code the way it probably should be) - if there is some bugs let me know and we can fix them 😄

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -203,7 +202,7 @@ const makeEditable = (row: Row, col: ColumnType) => {
 /** handle keypress events */
 const onKeyDown = async (e: KeyboardEvent) => {
   if (e.key === 'Alt') {
-    disableUrlOverlay.value = true
+    disableOverlay.value = true
```
Comment (dolsem) [inline packages/nc-gui/components/smartsheet/Grid.vue]: I am new to Vue, but wouldn't mutating a composable value like this be considered an anti-pattern? Makes it very hard to track where changes are coming from.

Comment (bcakmakoglu) [reply packages/nc-gui/components/smartsheet/Grid.vue]: Not really.

How would your implementation be easier to track? It does the exact same for the cost of more code and variables.

Comment (bcakmakoglu) [reply packages/nc-gui/components/smartsheet/Grid.vue]: I think the concept of composables needs a bit of clarification.
Composables aren't a rigid thing, but the core concept is obviously reusability of code.

Let's say we want to implement a simple toggle

```vue
<script lang="ts" setup>
const state = ref(false)

const toggleState = (nextState?: boolean) => (state.value = nextState ?? !state.value)
</script>

<template>
	<button @click="toggleState">Toggle</button>
	<div>{{ state ? 'toggled' : 'untoggled' }}</div>
</template>
```

In this example we can easily extract the functionality provided inside the setup function and make it reusable across any component for multiple purposes.

```ts
// useToggle.ts

function useToggle(initialState?: boolean) {
	const state = ref(initialState ?? false)
	
	const toggleState = (nextState?: boolean) => (state.value = nextState ?? !state.value)

	return [state, toggleState]
}
```

And now we can simply use our composable inside the component and save us the trouble of initializing refs etc. for the purpose of using a toggle

```vue
<script lang="ts" setup>
import { useToggle } from './useToggle'

const [state, toggleState] = useToggle()
</script>

<template>
	<button @click="toggleState">Toggle</button>
	<div>{{ state ? 'toggled' : 'untoggled' }}</div>
</template>
```

Simple enough.
`useToggle` will always initialize a new scope - meaning, every time you call it there will be a new ref for `state` and `toggleState` will always reference that `state` ref.
It is not considered an anti-pattern if you started to manipulate `state`.

`useCellUrlConfig` does not create a new scope every time it's called - because it is using `useInjectionState`.
Meaning, every time you call `useCellUrlConfig` there will only be a *single*  instance of it's scope that is available - never multiple.
Even if you called the `provide` function multiple times, it would only overwrite the state and not add a second one that could be used.
Basically like a Vuex or Pinia store. It's a single instance of a store to be used across a component tree, beginning from the point it was provided.

You're not wrong that it becomes harder to track where something was provided and possibly mutated, but using explicit `inject` and `provide` calls doesn't really remedy that situation, it just uses a different syntax to do the same thing 😄 

I also believe that, in the current state of the app, there is a lot of improvement in terms of setups and initialization so it becomes less troublesome to track values, which is something that will be worked on of course.

I hope this clears up why I decided to include the variables inside a single location instead of spreading it out into multiple locations.

Comment (dolsem) [reply packages/nc-gui/components/smartsheet/Grid.vue]: Thank you for the explanation

Comment (bcakmakoglu) [reply packages/nc-gui/components/smartsheet/Grid.vue]: If you're coming from a React background, don't confuse composables with React Hooks.

React Hooks are *similar* but composables are a lot more flexible and can basically be any type of functionality you want to re-use unlike React Hooks which, in most cases, *have* to be called inside a Functional Components' body.

**Comment by bcakmakoglu** [comment]
@dolsem if you're ok with this PR you are free to merge it btw 😄

**Comment by dolsem** [comment]
I haven't had the time to really check it yet, will try to get to it today. Thanks for the help :)

**Comment by dolsem** [comment]
@bcakmakoglu I was just testing it and the code is very broken. First of all, disabling overlay in Lookup disables overlay for all items, not just for descendants of the Lookup component. But even if I comment out the code in Lookup, the injection state still seems to be shared between items with different urls and it just doesn't work. I don't know how to fix it tbh in a way that you would want me to. Do you think we can just go back to the version on the base branch? I just tested it again and it works fine. Disabling overlay has a different data flow from applying url rules and trying to put them both inside one injection state results in logic that is unnecessarily convoluted imo.

**Comment by bcakmakoglu** [comment]
@dolsem Ah, I was under the impression that they were supposed to share a state, so I misunderstood your intended flow here, my bad :)
I reverted the injection changes, would you take another look if this is how you would want it to work?

**Comment by bcakmakoglu** [comment]
@dolsem 
I added my changes onto the latest dev - no more injection state for the composable, meaning there is no shared state for it and each invocation will have it's own computed values.
The injection for disable overlay stays.

Everything seems to be working as it did before.

Closing this as it's unnecessary now. Thanks for your time and contribution :)

**Comment by dolsem** [comment]
@bcakmakoglu looks like the version on the base branch got merged yesterday, using composable with injection state for overall cell url config, independent state for each url and the injection for disable overlay. Thanks for your input, composable with injection state was an elegant solution!
